src/test/java/parser/JsonParserTest.java

```diff
@@ -0,0 +1,52 @@
+package parser;
+
+import org.junit.jupiter.api.*;
+import shop.Cart;
+
+import java.io.File;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class JsonParserTest {
+    static File file;
+    @BeforeAll
+    static void fileCreation(){
+        file = new File("src/main/resources/raccoon-cart.json");
+    }
+
+    @Tag("happy")
+    @Test
+    void readFromFile() {
+        JsonParser parser = new JsonParser();
+        Cart cart = parser.readFromFile((new File("src/main/resources/eugen-cart.json")));
+
+        assertAll(
+                "group assertion for JsonParser read",
+                ()->assertNotNull(cart),
+                ()->assertEquals(cart.getClass(), Cart.class),
+                ()->assertEquals("eugen-cart", cart.getCartName())
```

Comment:
please provide a 3rd parameter - error message for eacg assertions

Reply:
Done